shared/trieutil/sparse_merkle.go

```diff
@@ -183,15 +185,14 @@ func (m *SparseMerkleTrie) ToProto() *protodb.SparseMerkleTrie {
 // VerifyMerkleBranch verifies a Merkle branch against a root of a trie.
 func VerifyMerkleBranch(root []byte, item []byte, merkleIndex int, proof [][]byte) bool {
 	node := bytesutil.ToBytes32(item)
-	currentIndex := merkleIndex
-	for i := 0; i < len(proof); i++ {
-		if currentIndex%2 != 0 {
+	for i := 0; i < int(params.BeaconConfig().DepositContractTreeDepth); i++ {
+		if (uint64(merkleIndex) / mathutil.PowerOf2(uint64(i)) % 2) != 0 {
 			node = hashutil.Hash(append(proof[i], node[:]...))
 		} else {
 			node = hashutil.Hash(append(node[:], proof[i]...))
 		}
-		currentIndex = currentIndex / 2
 	}
+	node = hashutil.Hash(append(node[:], proof[int(params.BeaconConfig().DepositContractTreeDepth)]...))
```

Comment:
Had to add this part, looking at how Root() function is calculation the Merkle root.

Would someone kind enough explain what is the purpose of `enc` below?:
```
func (m *SparseMerkleTrie) Root() [32]byte {
	enc := [32]byte{}
	binary.LittleEndian.PutUint64(enc[:], uint64(len(m.originalItems)))
	return hashutil.Hash(append(m.branches[len(m.branches)-1][0], enc[:]...))
}
```